examples/demos/example25-tree-view.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useState, useReducer, useMemo } from 'react';
+
+import DataGrid, { Column } from '../../src';
+import { CellExpanderFormatter, ChildRowDeleteButton } from './formatters';
+import Wrapper from './Wrapper';
+
+interface Row {
+  id: string;
+  name: string;
+  format: string;
+  position: string;
+  price: number;
+  children?: Row[];
+  parentId?: string;
+  isExpanded?: boolean;
+}
+
+interface Action {
+  type: 'toggleSubRow' | 'deleteSubRow';
+  id: string;
+}
+
+function createRows(): Row[] {
+  const rows = [];
+  for (let i = 0; i < 100; i++) {
+    const price = Math.random() * 30;
+    const id = `row${i}`;
+    const row: Row = {
+      id,
+      name: `supplier ${i}`,
+      format: `package ${i}`,
+      position: 'Run of site',
+      price,
+      children: [
+        { id: `${id}-0`, parentId: id, name: `supplier ${i}`, format: '728x90', position: 'run of site', price: price / 2 },
+        { id: `${id}-1`, parentId: id, name: `supplier ${i}`, format: '480x600', position: 'run of site', price: price * 0.25 },
+        { id: `${id}-2`, parentId: id, name: `supplier ${i}`, format: '328x70', position: 'run of site', price: price * 0.25 }
+      ],
+      isExpanded: false
+    };
+    rows.push(row);
+  }
+  return rows;
+}
+
+function toggleSubRow(rows: Row[], id: string): Row[] {
+  const rowIndex = rows.findIndex(r => r.id === id);
+  const row = rows[rowIndex];
+  const { children } = row;
+  if (!children) return rows;
+
+  const newRows = [...rows];
+  if (!row.isExpanded) {
+    newRows[rowIndex] = { ...newRows[rowIndex], isExpanded: true };
```

Comment:
```suggestion
    newRows[rowIndex] = { ...row, isExpanded: true };
```
We could also move this line out of the if/else, and do
```suggestion
    newRows[rowIndex] = { ...row, isExpanded: !row.isExpanded };
```

Reply:
Done